pimd/pim_igmpv2.c

```diff
@@ -140,6 +144,27 @@ int igmp_v2_recv_report(struct gm_sock *igmp, struct in_addr from,
 			   ifp->name, group_str);
 	}
 
+	/*
+	 * RFC 4604
+	 * section 2.2.1
+	 * EXCLUDE mode does not apply to SSM addresses, and an SSM-aware router
+	 * will ignore MODE_IS_EXCLUDE and CHANGE_TO_EXCLUDE_MODE requests in
+	 * the SSM range.
+	 */
+	if (pim_is_grp_ssm(pim_ifp->pim, group_addr)) {
+		if (PIM_DEBUG_IGMP_PACKETS) {
+			char grp_str[200];
+
+			inet_ntop(AF_INET, &group_addr, grp_str,
```

Comment:
Please use `zlog_debug("…%pI4…", … &group_addr …)` instead. This makes `grp_str` and `inet_ntop` unnecessary.

Reply:
Fixed